indigo_api/models/works.py

```diff
@@ -411,6 +444,82 @@ def save(self, *args, **kwargs):
+    provisions = JSONField(null=False, blank=False, default=list)
+
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    created_by_user = models.ForeignKey(User, on_delete=models.SET_NULL, null=True, related_name='+')
+    updated_by_user = models.ForeignKey(User, on_delete=models.SET_NULL, null=True, related_name='+')
+
+    class Meta:
+        ordering = ['date']
+        unique_together = (('commenced_work', 'commencing_work', 'date'),)
+
+    def is_only_commencement(self):
+        """ checks that the current commencement object is the only one associated with this commenced work,
+        as well as that it's the main one and that it commences all provisions.
+        """
+        return bool(
+            self.main is True and
+            self.all_provisions is True and
+            not self.other_commencements().exists()
+        )
+
+    def other_commencements(self):
+        return Commencement.objects.filter(commenced_work=self.commenced_work).exclude(pk=self.pk)
+
+    def save(self, *args, **kwargs):
+        # ensure only one commencement with main=True on commenced work
+        existing_main_commencement = self.commenced_work.main_commencement()
+        if existing_main_commencement and existing_main_commencement != self:
+            self.main = False
+
+        # ensure only one commencement with all_provisions=True on commenced work
+        existing_all_provisions_commencement = self.commenced_work.commencements.filter(all_provisions=True).first()
+        if existing_all_provisions_commencement and existing_all_provisions_commencement != self:
+            self.all_provisions = False
+
+        return super(Commencement, self).save(*args, **kwargs)
+
+
+class UncommencedProvisions(models.Model):
+    """ The details of uncommenced provisions of a work
+    """
+    commenced_work = models.OneToOneField(Work, on_delete=models.CASCADE, null=False, help_text="Principal work with uncommenced provisions", related_name="uncommenced_provisions")
+    all_provisions = models.BooleanField(default=False, help_text="All provisions of this work are uncommenced")
+
+    # list of the element ids of the uncommenced provisions, e.g. ["section-2", "section-4.3.list0.a"]
+    provisions = JSONField(null=False, blank=False, default=list)
+
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    created_by_user = models.ForeignKey(User, on_delete=models.SET_NULL, null=True, related_name='+')
+    updated_by_user = models.ForeignKey(User, on_delete=models.SET_NULL, null=True, related_name='+')
+
+    def save(self, *args, **kwargs):
+        # ensure only one commencement / uncommencement with all_provisions=True on commenced work
+        if self.all_provisions:
+            existing_all_provisions_commencement = self.commenced_work.commencements.filter(all_provisions=True).first()
```

Comment:
```suggestion
            existing_all_provisions_commencement = self.work.commencements.filter(all_provisions=True).exists()
```

Reply:
From a style standpoint, the long variable name may be descriptive but it's long to read and long to type. It could safely be `existing` or `commencement`.

Reply:
Faster to check `.exists()` than to load from the database.